src/client/jupyter/kernel-manager.ts

```diff
@@ -78,26 +78,25 @@ export class KernelManagerImpl extends EventEmitter {
         });
     }
 
-    public startKernelFor(language: string): Promise<Kernel> {
-        return this.getKernelSpecFor(language).then(kernelSpec => {
-            return this.startKernel(kernelSpec, language);
-        });
+    public async startKernelFor(language: string): Promise<Kernel> {
+        const kernelSpec = await this.getKernelSpecFor(language);
+        return this.startKernel(kernelSpec, language);
     }
 
     public startExistingKernel(language: string, connection, connectionFile): Promise<Kernel> {
         throw new Error('Start Existing Kernel not implemented');
     }
 
-    public startKernel(kernelSpec: KernelspecMetadata, language: string): Promise<Kernel> {
+    public async startKernel(kernelSpec: KernelspecMetadata, language: string): Promise<Kernel> {
```

Comment:
It would! The thing is I started with changes to symbol provider but soon discovered that the test suite no longer passes even on master. I'll try to split my changes into two PRs.